### PR TITLE
Diagnostic HUD Fixes and Upgrade

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This was originally going to just be a fix for diagnostic HUDs, but I noticed TG added a few cool features to them---and also that how we were handling diagnostic HUDs wasn't great. Credit to TG for most of this.

the fixes:
- When something gets added to a diagnostic HUD, it gets added to all of them now (basic and advanced) rather than just basic.
- Advanced diagnostic HUDs can now appropriately see a bot's travel path

Feature:
- Observers now get an advanced diagnostic HUD as opposed to basic
- AI gets an advanced diagnostic HUD instead of basic
- Augmented diagnostic eyes get an advanced diagnostic HUD instead of basic
- Diagnostic HUDs can now see which doors are electrified

Code stuff:
- standardized diagnostic HUD to /basic and /advanced, instead of the entry level one acting as a pseudo basic one

Pics:
img
img

:cl: Fox McCloud
fix: Advanced diagnostic HUDs can now see bot travel paths
add: AIs and observers and the diagnostic HUD augment now all have advanced diagnostic HUDs as opposed to basic ones
add: Diagnostic HUDs can see if a door is shocked or not
/:cl:

### CODE REVIEW

**Comment by Qwertytoforty** [comment]
Wow, this is nice!

**Comment by Mitchs98** [comment]
Turns a relatively useless item into something neat and nice. Looks cool~

**Comment by taukausanake** [comment]
This is making the diag HUD a lot more machine/electronics based rather than just robotics. I like it but I'm wondering if there should be some separation of the two now. I could see this picking up more information on other machines, like anything that is hacked or other information on doors (whenever power or AI control has been cut). As #13769 has shown there are also some machines that can also be electrified so might not be a bad idea to look to in a future PR

**Comment by MRgaty** [comment]
The icon for electrified doors looks pretty weird tbh, I couldnt express why tho.
Maybe an artist could tell better if they find it as weird and out of place.

**Comment by taukausanake** [comment]
The corners are not pointed enough and the center bit isn't tangent to itself. Kind of makes it look not straight

File: code/modules/mob/living/silicon/silicon.dm [hunk 1]
```diff
@@ -33,15 +33,15 @@
 
 	var/med_hud = DATA_HUD_MEDICAL_ADVANCED //Determines the med hud to use
 	var/sec_hud = DATA_HUD_SECURITY_ADVANCED //Determines the sec hud to use
-	var/d_hud = DATA_HUD_DIAGNOSTIC_ADVANCED //There is only one kind of diag hud
+	var/d_hud = DATA_HUD_DIAGNOSTIC_BASIC //There is only one kind of diag hud
```
Comment (AffectedArc07) [inline code/modules/mob/living/silicon/silicon.dm:36]: Comment is inaccurate but I am not having this merge-blocked over that

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -446,3 +448,13 @@
...
+
+/*~~~~~~~~~~~~
+	Airlocks!
+~~~~~~~~~~~~~*/
+/obj/machinery/door/airlock/proc/diag_hud_set_electrified()
+	var/image/holder = hud_list[DIAG_AIRLOCK_HUD]
+	if(isElectrified())
+		holder.icon_state = "electrified"
+	else
+		holder.icon_state = ""
```
Comment (farie82) [inline code/game/data_huds.dm:460]: Small nitpick. Normally it'll be set to null instead of an empty string

**Review by farie82 — APPROVED** [review]

**Pull request merged**